src/init.cpp

```diff
@@ -1569,34 +1568,25 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
                                      chainman, *node.mempool, ignores_incoming_txs);
     RegisterValidationInterface(node.peerman.get());
 
-    // ********************************************************* Step 8: start indexers
-
-    // If reindex-chainstate was specified, delay syncing indexes until ImportBlocks has reindexed the chain
-    if (!fReindexChainState) g_indexes_ready_to_sync = true;
+    // ********************************************************* Step 8: start indexes
     if (args.GetBoolArg("-txindex", DEFAULT_TXINDEX)) {
         auto result{WITH_LOCK(cs_main, return CheckLegacyTxindex(*Assert(chainman.m_blockman.m_block_tree_db)))};
         if (!result) {
             return InitError(util::ErrorString(result));
         }
 
         g_txindex = std::make_unique<TxIndex>(interfaces::MakeChain(node), cache_sizes.tx_index, false, fReindex);
-        if (!g_txindex->Start()) {
```

Comment:
In commit "init: start indexes after 'loadblk' thread finishes" (a9e45f7926fab62ef0c5cea540386266235b2990)

Can the commit message be written to say how this affects behavior of bitcoind?

 I understand that it gets rid of the `g_indexes_ready_to_sync` variable internally, which is nice.

I can also see that behavior of the `BaseIndex::ThreadSync()` is basically unchanged, because it now calls `BaseIndex::Start` in basically same place it previously set `g_indexes_ready_to_sync` to true, after importing blocks and before loading the mempool.

But now `RegisterValidationInterface` and `BaseIndex::Init` calls appear to be delayed. Previously they ran on the main thread before loadblk thread started, now they are running **on** the loadblk thread, after importing blocks and before loading the mempool. Is it a good thing to delay the index Register and Init calls? Is it good to move them from the main thread to the loadblk thread? It seems like it would be pretty easy to keep those calls where they are and still get rid of the `g_indexes_ready_to_sync` variable, so what is the motivation for changing the behavior here?

The change seems reasonable to me, but the description is vague and it's not clear how exactly behavior is changing and why the behavior change is good, or neccessary.

EDIT: Looking at later commit "refactor: index, decouple 'Init' from 'Start'" (9f38c2ffba87ea07c0eeb9975ad4d84ae716f9aa), I can see that the behavior change here actually gets reverted later and is unnecessary. I would suggest moving that commit before this one so this one only removes the `g_indexes_ready_to_sync` variable without affecting other behavior.